src/App.tsx

```diff
@@ -15,49 +17,91 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortField {
+  LENGTH = 'length',
+  ALPHABETICALLY = 'alphabet',
+}
+
+function getPrepared(goods: string[], sortField: string, isReversed: boolean) {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SortField.LENGTH:
+        preparedGoods
+          .sort((good1: string, good2: string) => good2.length - good1.length);
+        break;
+
+      case SortField.ALPHABETICALLY:
+        preparedGoods
+          .sort((good1: string, good2: string) => good1.localeCompare(good2));
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  if (isReversed) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, SetIsReversed] = useState(false);
+
+  const visibleGoods = getPrepared(goodsFromServer, sortField, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames('button', 'is-info',
+            { 'is-light': sortField !== SortField.ALPHABETICALLY })}
+          onClick={() => setSortField(SortField.ALPHABETICALLY)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames('button', 'is-success',
+            { 'is-light': sortField !== SortField.LENGTH })}
+          onClick={() => setSortField(SortField.LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames('button', 'is-warning',
+            { 'is-light': !isReversed })}
+          onClick={() => {
+            SetIsReversed(!isReversed);
+            getPrepared(goodsFromServer, sortField, isReversed);
+          }}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortField || isReversed) && (
+          <button
+            type="button"
+            className={classNames('button', 'is-danger', 'is-light')}
```

Comment:
```suggestion
            className={'button is-danger is-light'}
```
you don't have the condition here